src/lib/hooks/useConnectionGun.js

```diff
@@ -0,0 +1,135 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { AppState } from 'react-native'
+import Config from '../../config/config'
+import logger from '../logger/pino-logger'
+import SimpleStore from '../undux/SimpleStore'
+
+let isFirstCheckGun = false
+let needToBindEventsGun = true
+
+const log = logger.child({ from: 'useHasConnectionGun' })
+
+export default () => {
+  const [isConnection, setIsConnection] = useState(true)
+  const store = SimpleStore.useStore()
+  const userStorage = store.get('userStorage')
+  const connectionCheck = useRef()
+
+  /**
+   * Kills the last connection listener alive.
+   */
+  const killLastConnectionCheck = () => {
+    if (typeof connectionCheck.current === 'function') {
+      connectionCheck.current()
+      connectionCheck.current = null
+    }
+  }
+
+  /**
+   * isGunConnection gets recursive when the connection fails, so it returns a function to kill
+   * the recursion.
+   */
+  const isGunConnection = useCallback(() => {
+    killLastConnectionCheck()
```

Comment:
can you explain why this happens?
we use setTimeout, which should be invoked once, what you are saying is that there is a problem where isGunConnection is invoked multiple times and not via the timer

Reply:
Every time the app changes its state to active, isGunConnection will be called.  It's also colled by gunClose & gunError.

When isGunConnection fails to connect it will call itself again in the next second. (isGunConnection -> setTimeout -> isGunConnection)

We need to stop the loop every time we call isGunConnection again from outside of the loop (app state change, gunClose, gunError).

Reply:
@yaroslav-fedyshyn-nordwhale please apply your changes from 
https://github.com/GoodDollar/GoodDAPP/pull/1429
also here
and then make a PR to master cherrypicking these commits

Reply:
@sirpy 

ok.
#1429 is already PR to master

Reply:
> @yaroslav-fedyshyn-nordwhale please apply your changes from
> #1429
> also here
> and then make a PR to master cherrypicking these commits

Done